sdk/dynamicconfig/metric/controller/notify/mock/notifier.go

```diff
@@ -0,0 +1,43 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mock
+
+import (
+	"go.opentelemetry.io/contrib/sdk/dynamicconfig/metric/controller/notify"
+)
+
+type Notifier struct {
+	data chan *notify.MetricConfig
+}
+
+func NewNotifier() *Notifier {
+	return &Notifier{make(chan *notify.MetricConfig)}
+}
+
+func (n *Notifier) Receive(config *notify.MetricConfig) {
```

Comment:
Where is this used?

Reply:
It's used mainly in the [push_test](https://github.com/vmingchen/opentelemetry-go-contrib/blob/205cb1ad81876cd9d1db3b1c9d8c1432842f1a96/sdk/dynamicconfig/metric/controller/push/push_test.go#L332) as a way to mock the process of "receiving" an updated metric config.